From fc6413775e8fd011f9583280f30f7c46defa6723 Mon Sep 17 00:00:00 2001
From: Sami Kerola <kerolasa@iki.fi>
Date: Wed, 18 Jul 2012 21:27:42 +0200
Subject: [PATCH 06/44] Disallow use of unset variables

Use of unset variable should be considered as bug, which means there
has to be sensible defaults.  For $skip_qemu_kvm the default, or later
setting, is a binary named 'true' or 'false' which is executed at it
clause.

Signed-off-by: Sami Kerola <kerolasa@iki.fi>
(cherry picked from commit a1670e2cb0ec7fd4116cba9c634ba2c92cb3f7e4)
---
 virt-what.in | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/virt-what.in b/virt-what.in
index d50426a..d62f7c8 100644
--- a/virt-what.in
+++ b/virt-what.in
@@ -26,6 +26,11 @@
 # The following resources were useful in writing this script:
 # . http://www.dmo.ca/blog/20080530151107
 
+# Do not allow unset variables, and set defaults.
+set -u
+root=''
+skip_qemu_kvm=false
+
 VERSION="@VERSION@"
 
 function fail {
@@ -171,14 +176,14 @@ fi
 # Check for Parallels.
 if echo "$dmi" | grep -q 'Vendor: Parallels'; then
     echo parallels
-    skip_qemu_kvm=1
+    skip_qemu_kvm=true
 fi
 
 # Check for Xen.
 
 if [ "$cpuid" = "XenVMMXenVMM" ]; then
     echo xen; echo xen-hvm
-    skip_qemu_kvm=1
+    skip_qemu_kvm=true
 elif [ -f $root/proc/xen/capabilities ]; then
     echo xen
     if grep -q "control_d" $root/proc/xen/capabilities; then
@@ -186,7 +191,7 @@ elif [ -f $root/proc/xen/capabilities ]; then
     else
         echo xen-domU
     fi
-    skip_qemu_kvm=1
+    skip_qemu_kvm=true
 elif [ -f $root/sys/hypervisor/type ] &&
     grep -q "xen" $root/sys/hypervisor/type; then
     # Ordinary kernel with pv_ops.  There does not seem to be
@@ -217,7 +222,7 @@ fi
 # seen that it's Parallels.  Xen uses QEMU as the device model, so
 # skip this test if we know it is Xen.
 
-if [ ! "$skip_qemu_kvm" ]; then
+if ! $skip_qemu_kvm; then
     if [ "$cpuid" = "KVMKVMKVM" ]; then
 	echo kvm
     else
-- 
1.8.3.1

